stdlib/public/stubs/CommandLine.cpp

```diff
@@ -104,6 +104,42 @@ extern "C" char ** _swift_stdlib_getUnsafeArgvArgc(int *outArgLen) {
   *outArgLen = __argc;
   return __argv;
 }
+#elif defined(__FreeBSD__)
+#include <errno.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/param.h>
+#include <sys/sysctl.h>
+
+SWIFT_RUNTIME_STDLIB_INTERFACE
+extern "C" char ** _swift_stdlib_getUnsafeArgvArgc(int *outArgLen) {
+  assert(outArgLen != nullptr);
+
+  if (_swift_stdlib_ProcessOverrideUnsafeArgv) {
+    *outArgLen = _swift_stdlib_ProcessOverrideUnsafeArgc;
+    return _swift_stdlib_ProcessOverrideUnsafeArgv;
+  }
+
+  char argPtr[8192]; // or use ARG_MAX? 8192 is used in LLDB though..
```

Comment:
Sweet.

In terms of buffer size, 8192 is probably fine, and ARG_MAX (256 KiB) is probably too large for a stack allocation.

For something that's guaranteed to work with ridiculously large argv arrays, the following would do the trick:

- Fetch the current size (by providing a NULL oldp argument).
- Allocate your buffer on the heap, and then call sysctl again to populate the buffer.
- If you get a return value of ENOMEM with an oldlenp equal to your buffer size, you lost a race with someone else setting KERN_PROC_ARGS -- you can safely loop, grow the buffer, and try again.